documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -93,225 +66,84 @@ include::config_examples/serialization_marshaller.xml[]
+to using JBoss Marshalling.
 
-----
-java.io.NotSerializableException: java.lang.Object
-at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:857)
-at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:407)
-at org.infinispan.marshall.exts.ReplicableCommandExternalizer.writeObject(ReplicableCommandExternalizer.java:54)
-at org.infinispan.marshall.jboss.ConstantObjectTable$ExternalizerAdapter.writeObject(ConstantObjectTable.java:267)
-at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:143)
-at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:407)
-at org.infinispan.marshall.jboss.JBossMarshaller.objectToObjectStream(JBossMarshaller.java:167)
-at org.infinispan.marshall.VersionAwareMarshaller.objectToBuffer(VersionAwareMarshaller.java:92)
-at org.infinispan.marshall.VersionAwareMarshaller.objectToByteBuffer(VersionAwareMarshaller.java:170)
-at org.infinispan.marshall.DefaultMarshallerTest.testNestedNonSerializable(VersionAwareMarshallerTest.java:415)
-Caused by: an exception which occurred:
-in object java.lang.Object@b40ec4
-in object org.infinispan.commands.write.PutKeyValueCommand@df661da7
-... Removed 22 stack frames
-----
-
-The way the "in object" messages are read is the same in which stacktraces are read. The highest "in object" being the most inner one and the lowest "in object" message being the most outer one. So, the above example indicates that a java.lang.Object instance contained in an instance of org.infinispan.commands.write.PutKeyValueCommand could not be serialized because java.lang.Object@b40ec4 is not serializable.
-
-This is not all though! If you enable DEBUG or TRACE logging levels, marshalling exceptions will contain show the toString() representations of objects in the stacktrace. For example:
-
-----
-java.io.NotSerializableException: java.lang.Object
-...
-Caused by: an exception which occurred:
-in object java.lang.Object@b40ec4
--> toString = java.lang.Object@b40ec4
-in object org.infinispan.commands.write.PutKeyValueCommand@df661da7
--> toString = PutKeyValueCommand{key=k, value=java.lang.Object@b40ec4, putIfAbsent=false, lifespanMillis=0, maxIdleTimeMillis=0}
-----
+WARNING: link:{javadocroot}/org/infinispan/commons/marshall/AdvancedExternalizer.html[AdvancedExternalizer] implementations
+are ignored when persisting data unless JBoss marshalling is configured. However, we strongly recommend migrating away
+from AdvancedExternalizers and JBoss marshalling as these are deprecated and will be removed in a future version
+of {brandname}.
 
-With regards to unmarshalling exceptions, showing such level of information it's a lot more complicated but where possible. {brandname} will provide class type information. For example:
+=== Custom Implementation
+It's also possible for users to provide their own link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html[marshaller implementation],
```

Comment:
Might be nice to address readers directly. You can provide custom marshaller classes with the link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html[Marshaller interface].